### ISSUE TITLE
Questions for implement.

### ISSUE THREAD

**Issue by PCH10507323** [bingykang/Fewshot_Detection#5]
Hi, thanks for your sharing, but I have two questions.
1. In training phase, we'll concat masks and images before input them to meta model. However, the mask information is come from ground truth label, and we won't have it in testing phase. So when we testing, the inputs for the meta model are totally different, how can we solve this or I have any misunderstanding?

2. In testing phase, after we got N set of reweighting coefficients, how to know which coefficients in the N set should we use for the testing sample?

Hope you can help me to clarify this question, thanks.

**Comment by bingykang** [comment]
Hi, thank you for your interest. 

1. The reweighting coefficients are the key here. At testing, the coefficients are actually from the training set. Each class only have one reweighting vector by averaging all training examples within this category.  The meta-model is only used at training to learn such coefficients, but it can be discarded at testing as long as we compute reweighting vectors from the training set. 

2. I guess N means N classes here? If so, we use them all.

**Comment by PCH10507323** [comment]
Hi, thanks for your quickly reply.
The first question I can understand now.
For the second one, my question is after we get N set of reweighting coefficients (N is class nums) from all training images, we need to use each of them to multiply with the output vectors from extractor, so each test image will have N set of output vector? If so, how do we decide which one is the final detection result in the N set of outputs? Thanks.

**Comment by bingykang** [comment]
We use them all, each is responsible for generating predictions for that class. You can refer to the paper for details.

**Comment by PCH10507323** [comment]
Thanks, let me check details first.
